rust/tvm-rt/src/to_function.rs

```diff
@@ -46,46 +46,52 @@ pub use tvm_sys::{ffi, ArgValue, RetValue};
 /// And the implementation of it to `ToFunction`.
 pub trait Typed<I, O> {
     fn args(i: &[ArgValue<'static>]) -> Result<I>;
-    fn ret(o: O) -> RetValue;
+    fn ret(o: O) -> Result<RetValue>;
 }
 
-impl<F, O: Into<RetValue>> Typed<(), O> for F
+impl<F, O, E> Typed<(), O> for F
 where
     F: Fn() -> O,
+    Error: From<E>,
```

Comment:
Does this add a bound to TryInto<... Error> ?

Reply:
This allows us to convert the Result<T, E> into Result<T, Error> below.